test/unit/core.js

```diff
@@ -1217,6 +1217,25 @@ test("jQuery.each(Object,Function)", function() {
 	equal( i, 2, "Iteration over document.styleSheets" );
 });
 
+test( "JIT compilation does not interfere with length retrieval (gh-2145)", function() {
+	expect( 4 );
+
+	var i;
+
+	// Trigger JIT
```

Comment:
Do you mean more specific about how this triggers JIT compilation or which function is being JIT'd? Perhaps something like, "Trigger JIT compilation of `jQuery.each` – and therefore `isArraylike` – in iOS"?